### PR TITLE
Add support for log streams

### PR DESCRIPTION
**Pull request by maybegreat48**

### CODE REVIEW

File: src/LogStream.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		
+		const std::string& Name() const;
+		const void* UserData() const;
+		bool Enabled() const;
+
+		void SetEnabled(bool state);
+
+	private:
+		std::string m_Name;
+		void* m_UserData;
```
Comment (Yimura) [inline src/LogStream.hpp]: What's the usage of this?

**Review by Yimura — CHANGES_REQUESTED** [review]
Mixed indents here and there.

Could you give an example of how usage of this feature would look like in-code?

File: src/LogStream.hpp (hunk 1 above)
Comment (maybegreat48) [reply src/LogStream.hpp]: Clients can add an userdata (pointer to anything) that they can later retrieve at the sink

File: include/AsyncLogger/Logger.hpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #pragma once
-#if !defined(USE_FMT) && CXX_FORMAT_SUPPORT
+#if !defined(USE_FMT) && CXX_FORMAT_SUPPORT || true
```
Comment (xiaoxiao921) [inline include/AsyncLogger/Logger.hpp]: Such hack not needed when add_compile_definitions(CXX_FORMAT_SUPPORT) is properly defined

**Review by xiaoxiao921 — CHANGES_REQUESTED** [review]

**Pull request merged**